### DISCUSSION TITLE
I’m looking for a maintainer

### DISCUSSION THREAD

**Discussion by WeeJeWel** [wg-easy/wg-easy#553] [Announcements]
**Update Dec. 21 2023: @pheiduck and @peterlewis are now taking over as maintainers. Thank you! 🎉**

This project has reached more users than I hoped, yet I do not have the time available to maintain this project sufficiently.

While I can consider it finished, there are a few features I’d like to see:

* Seamless fallback support for older kernels that don’t have WireGuard built in. This should work automatically without user invention, just a console log to inform the user.
* Sessionless HTTP API authentication by using the WG_PASSWORD in the Authorization header.
* A one-shot tool/script to convert an existing WireGuard .conf file to ‘wg0.json’ for people that would like to migrate to wg-easy.

I find it extremely important that this project stays simple and contained. A future maintainer should share this philosophy and rigorously close any PRs related to built-in support for other tools, features for a minority, configuration options for running out-of-Docker, etc.

People like wg-easy because it’s easy and simple. The nature of open source is entropy and feature creep. It’s hard but liberating to say no to most contributions. Luckily everyone can fork so you never have to feel bad :)

Let me know if you’re interested in becoming a maintainer by creating a great PR for one of the above todos.

**Comment by pheiduck** [comment]
Will take over this job with @peterlewis ;)

**Reply by Hansaplast27** [reply]
Wow, I feel so many xmas vibes, its all coming together ;-). 

I do fully support WeejeWel and now hopefully @pheiduck and @peterlewis to KISS this repo. It works, and now is gonna get security updates too. Thats all I need from my home-hosted TailScale.

**Reply by pheiduck** [reply]
@Hansaplast27 I merge v12 on my fork tomorrow which have dark mode and sessionless HTTP API authentication

https://github.com/pheiduck/wg-easy/pull/45

**Reply by peterlewis** [reply]
I would be delighted to second @pheiduck's proposal and to take on co-responsibility of maintaining your origin repo! As I mentioned in reply on the issue, there's nothing I would love more than to put my fork into archive mode and for all efforts to be focused upstream.

It would be great to hop on an email/call/Jitsi to fully clarify your desires with regard to the breadth of future updates. I believe that neither @pheiduck or I would like to see any future iteration become over complicated and burdensome, but would be great to clarify, more precisely, the desired onward trajectory.

Sound good? 😃

**Reply by pheiduck** [reply]
@peterlewis Yes of course if we hold a meeting we just need to find a date where everyone can otherwise by e-mail.

**Reply by pheiduck** [reply]
But apart from which code changes flow into the main project, this requires a lot of rebase. 😬

**Reply by pheiduck** [reply]
> Seamless fallback support for older kernels that don’t have WireGuard built in. This should work automatically without user invention, just a console log to inform the user.

Is that really necessary? Since alpine runs as its own system with the current kernel (Alpine 3.19, Kernel 6.6 LTS)

**Reply by peterlewis** [reply]
> But apart from which code changes flow into the main project, this requires a lot of rebase. 😬
> 
> 

Absolutely. Needs considered thought! Ultimately our fork(s) differ from upstream in 3 main areas:

- Versioning bumps (e.g security fixes)
- Enhancements (e.g. exposing additional variables)
- Features (e.g. dark mode, stateless auth).

Of course the version bumps are a must, but with regards to enhancements and features, I suggest we put these areas to @WeeJeWel and ask which areas/specifics are desired upstream.

Would also be worth thinking about whether, following this, we should release one final update to our respective forks that aims to guide users back upstream with a message in the web console...?

In terms of availability for a meeting, I'm very flexible so happy to be led by @pheiduck and @WeeJeWel.

Mine and Emile's email addresses are on our GitHub profiles, so Philip, please can you send an email to us both and we can go from there? Thanks!

Peter 😀

**Reply by WeeJeWel** [reply]
FYI I've updated the workflows to publish to GHCR instead of DockerHub. Makes permissions easier too.

Also updated to Node.js v20 and updated the dependencies.

**Reply by pheiduck** [reply]
The bump to Node.js v20 was a bit to early because there are some issues on armv6/7 while build. See: https://github.com/nodejs/docker-node/issues/1946

**Reply by pheiduck** [reply]
As I saw on my docker hub mostly x64 and aarch64 is in use I will drop armv6/7 later next year.

**Reply by peterlewis** [reply]
So as to make life simpler, I have merged in your changes, @pheiduck (thank you) so that there's no disparity between our forks. Keen to make life as simple and straightforward for users of wg-easy, whilst we still have three different repos that are actively used.

**Comment by WeeJeWel** [comment]
> > Seamless fallback support for older kernels that don’t have WireGuard built in. This should work automatically without user invention, just a console log to inform the user.
> 
> Is that really necessary? Since alpine runs as its own system with the current kernel (Alpine 3.19, Kernel 6.6 LTS)

Because WireGuard is kernel-based, hosts with an older kernel don't work. We should embed wireguard-go (see https://www.wireguard.com/xplatform/) for those hosts.

**Reply by peterlewis** [reply]
Amazing news! I know @pheiduck experienced some funkiness with the build when he was testing v20, so this is great news.

Really look forward to archiving my fork and help people swim people back upstream, so to speak! :)

**Reply by WeeJeWel** [reply]
Indeed, Node.js v20 hangs on building. Just downgraded to v18 and that works perfectly.

I wouldn't want to drom armv6/armv7 support soon, supporting all those old Raspberry Pis out there is nice :)

**Reply by peterlewis** [reply]
Ah that helps to clarify. Thank you! Was certainly surprised it had suddenly resolved itself! :) @pheiduck are you happy to ping us both an email, at your convenience?

**Reply by pheiduck** [reply]
> Ah that helps to clarify. Thank you! Was certainly surprised it had suddenly resolved itself! :) @pheiduck are you happy to ping us both an email, at your convenience?

@WeeJeWel and @peterlewis: I shared my Email to you via Mail by trust, please don't share it with others. Thank you! 🙏

**Comment by pheiduck** [comment]
@WeeJeWel I would like to thank you again for giving me and Peter the confidence to continue working on this project.
I wish you and your family a Merry Christmas and a Happy New Year. 🙂
